src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState_SourceGenerators.cs

```diff
@@ -38,7 +44,11 @@ private sealed record SourceGeneratorMap(
     /// process (if present) and having it make the determination, without the host necessarily loading generators
     /// itself.
     /// </summary>
-    private static readonly ConditionalWeakTable<ProjectState, AsyncLazy<bool>> s_hasSourceGeneratorsMap = new();
+    private static readonly Dictionary<string, AnalyzerReferenceMap> s_languageToAnalyzerReferenceMap = new()
+    {
+        { LanguageNames.CSharp, new() },
+        { LanguageNames.VisualBasic, new() },
```

Comment:
it's fine to have these be keyed just by VB/C#.  OOP is only supported for those two languages alone.  This makes it easy to initialize and look into this without locking, or complex types like ConcurrentDict.